buddy_init/init_actions.py

```diff
@@ -0,0 +1,95 @@
+import os
+import pathlib
+import shutil
+import subprocess
+import venv
+from distutils import dir_util
+
+import fabric
+import paramiko
+from git import Repo
+
+from .init_framework import FatalException
+
+
+def _is_venv():
+    return bool(os.environ.get('VIRTUAL_ENV'))
+
+
+def _call_python(command: str, use_local_venv=True):
+    if use_local_venv:
+        subprocess.check_output(f'venv/bin/python3 -m {command}', shell=True)
+    elif _is_venv():
+        subprocess.check_output(f'python3 -m {command}', shell=True)
+    else:
+        raise FatalException('Installing packages on the system python is not allowed. Aborting')
+
+
+def create_venv():
+    """Create the experiment venv, otherwhise use the already active venv"""
+    venv.EnvBuilder(with_pip=True).create('venv')
+    subprocess.check_output(f'venv/bin/python3 -m pip install --upgrade pip', shell=True)
+    print(f"""\n\nRemember to source your new environment with:
+        source {os.getcwd()}/venv/bin/activate\n\n""")
+
+
+def create_git_repo():
+    """Setups the git repo with the proper git ignores"""
+    Repo.init('./.git', bare=True)
+
+
+def create_base_structure(use_local_venv=True):
```

Comment:
this should be:
git clone --depth=1 git://github.com/ministry-of-silly-code/examples/ examples
rm -rf ./examples/.git

not sure about the exact syntax.

That would avoid having to duplicate the code

Reply:
The example project is in the init (at least for now) as said in our calls

Reply:
That makes sense.
I'm still uneasy about a project within a project but deprecating the example repo could be a way.

Does this handle the need for a remote example repo on github (formely a fork of examples)?

Reply:
It creates the repo, but it does not pushes (it throws a warning in that case) the user still has to create the remote on its github

Reply:
without a push the example will fail and be useful
the user should create the example repo during the init (if he wants to initialize the example)
and we should add that as remote
e.g.
Do you want to create an example project [y,n]: y
Create a github repo and paste the ssh address, it looks like: git@github.com:username/example.git:

Reply:
Done, but I had to fight badly with GitPyhon (due to the path mismatch between where git is called and the repo is), in the end I had to go brutally with bash